### DISCUSSION TITLE
Generated GradCAM is not as expected

### DISCUSSION THREAD

**Discussion by MJGroeneveld** [Project-MONAI/MONAI#6747] [Q&A]
Hi! 

I use GradCAM from monai.visualize and implemented it into my pytorch lightning architecture. However, I get something different than that I was expecting: 

![image](https://github.com/Project-MONAI/MONAI/assets/79461911/8c7607c2-d757-43d0-acdd-148ca64a4509)

I would expect that the high activation areas (indicated by hot colors) corresponding to the regions that are most relevant for predicting. However, it is now blue instead of red. 

This is a part of my code where I generate the heatmaps. 

` class ClassifierChest(pl.LightningModule): 
def __init__(self, *args): 
        super().__init__() 

        #defining model 
        self.model_name = config['model_name']
        self.model = create_model(self.model_name)

       #defining GradCAM
        self.gradcam = GradCAM(
            nn_module=self.model,
            target_layers="model.class_layers.relu"
        )
def test_step(self, batch, batch_idx):
        X, y_class, _, image_names  = batch
        X, y_class                  = X.float().to(device), y_class.to(device)
        y_hat_class                 = self.model(X).squeeze(1)
        y_prob                      = torch.sigmoid(y_hat_class)
        test_loss                   = F.binary_cross_entropy_with_logits(y_hat_class, y_class.float())

       grayscale_cams = self.gradcam(x=X, class_idx=None)
       for image, grayscale_cam, pred_class, label, image_id in zip(X, grayscale_cams, y_prob, y_class, image_names): 
                    label = label.cpu().numpy()
                    pred_class = pred_class.cpu().numpy()

                    # Add GradCAM to image
                    heatmap, result = visualize_cam(grayscale_cam, image)

                    # Create figure and axes
                    fig, axes = plt.subplots(1,2)
                    image_normalized = (image - image.min()) / (image.max()-image.min())
                    image_normalized = np.clip(image_normalized.cpu(), 0, 1)
                    axes[0].imshow(image_normalized.permute(1, 2, 0))
                    result_normalized = (result - result.min()) / (result.max() - result.min())
                    result_normalized = np.clip(result_normalized, 0, 1)
                    axes[1].imshow(result_normalized.permute(1, 2, 0))`

and the model: 

`import pytorch_lightning as pl
import torchvision
import torch.nn as nn
import torch 
from monai.networks.nets import DenseNet121

class DenseNet(pl.LightningModule): 
    def __init__(self): 
        super().__init__() 
        self.model = DenseNet121(spatial_dims=2, in_channels=3, out_channels=1, pretrained=True)
    def forward(self, x): 
        x = self.model(x)
        return x `

**Comment by KumoLiu** [comment]
Hi @MJGroeneveld, thanks for your interest here.
Did you look at the logits in the section where you plot the activation map? I believe the color is related to the logits value and also reflects where your network is focused.

Hope it helps, thanks!

**Reply by emsdiaz** [reply]
Hi @KumoLiu I'm encountering a similar issue in my problem and noticed the same in this tutorial: [Class_lung_lesion.ipynb](https://github.com/Project-MONAI/tutorials/blob/main/modules/interpretability/class_lung_lesion.ipynb ). Could you help me understand why there is an inversed relevance (at least compared to what is normally expected) in the Grad-CAMs generated with MONAI?
